### DISCUSSION TITLE
Cask

### DISCUSSION THREAD

**Discussion by TangChiaHsin** [Homebrew/discussions#4352] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.9
ORIGIN: https://mirrors.ustc.edu.cn/brew.git
HEAD: 584588afb4ab557bcc0536f2794b4d043c333bb9
Last commit: 3 days ago
Core tap origin: https://mirrors.ustc.edu.cn/homebrew-core.git
Core tap HEAD: b7e525330968bcfe754ea1dd95b3f4768feaa0a0
Core tap last commit: 2 days ago
Core tap branch: master
Core tap JSON: 22 Mar 15:46 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_API_DOMAIN: https://mirrors.ustc.edu.cn/homebrew-bottles/api
HOMEBREW_BOTTLE_DOMAIN: https://mirrors.ustc.edu.cn/homebrew-bottles
HOMEBREW_BREW_GIT_REMOTE: https://mirrors.ustc.edu.cn/brew.git
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://mirrors.ustc.edu.cn/homebrew-core.git
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

input: `brew install mactex`
output:
==> Caveats
You must restart your terminal window for the installation of MacTex CLI tools to take effect.
Alternatively, Bash and Zsh users can run the command:

  eval "$(/usr/libexec/path_helper)"

==> Downloading https://mirror.ctan.org/systems/mac/mactex/mactex-20220321.pkg
==> Downloading from https://mirrors.sustech.edu.cn/CTAN/systems/mac/mactex/mactex-20220321.pkg
curl: (22) The requested URL returned error: 404

Error: Download failed on Cask 'mactex' with message: Download failed: https://mirror.ctan.org/systems/mac/mactex/mactex-20220321.pkg

It should be 404 not found certainly. Because the latest mactex version is **2023.0314**, not **2022.0321**.
So plz update the script, I'm waiting for using it.🤣

### Relevant casks

mactex

**Comment by gromgit** [comment]
https://github.com/Homebrew/homebrew-cask/pull/143411#issuecomment-1481066427
